NugetLightRequest.cs

```diff
@@ -1909,5 +1930,169 @@ private IEnumerable<PackageItem> SearchForPackages(PackageSource source, string
                 return Enumerable.Empty<PackageItem>();
             }
         }
+
+        internal NetworkCredential GetCredsFromCredProvider(string query, NuGetRequest request, bool isRetry=false)
+        {
+            if (query.IsNullOrEmpty())
+            {
+                throw new ArgumentNullException("query");
+            }
+
+            string username = "";
+            string password = "";
+            // Find credential provider
+            // Option 1. Use env var 'NUGET_PLUGIN_PATHS' to find credential provider
+            // see: https://docs.microsoft.com/en-us/nuget/reference/extensibility/nuget-cross-platform-plugins#plugin-installation-and-discovery
+            // Note: OSX and Linux can only use option 1
+            string credProviderPath = "";
+            string defaultEnvPath = "NUGET_PLUGIN_PATHS";
+            string nuGetPluginsPathNetCore = "%UserProfile%/.nuget/plugins/netcore/CredentialProvider.Microsoft/CredentialProvider.Microsoft.exe";
+            string nugetPluginsPathNetFramework = "%UserProfile%/.nuget/plugins/netfx/CredentialProvider.Microsoft/CredentialProvider.Microsoft.exe";
+            string nugetPluginPath = Environment.GetEnvironmentVariable(defaultEnvPath);
+            string fullNuGetPluginsPathNetCore = Environment.ExpandEnvironmentVariables(nuGetPluginsPathNetCore);
+            string nuGetPluginsPathNetFramework = Environment.ExpandEnvironmentVariables(nugetPluginsPathNetFramework);
+            if (!nugetPluginPath.IsNullOrEmpty())
```

Comment:
consider: this might be a bit cleaner if you had something like:
if(OSInformation.IsWindowsPowerShell) { pluginPath = "%UserProfile%...netcore" } else { pluginPath = "%..netfx" }
path = ExpandEnvironmentVariables(path)
if(File.Exists(path)) { credProviderPath = path}